### PR TITLE
fix: support ABI3 on Windows

### PR DESCRIPTION
**Pull request by henryiii**
Fixing ABI3 support.

### CODE REVIEW

File: cmake/nanobind-config.cmake [hunk 1]
```diff
@@ -195,7 +195,11 @@ endfunction()
 
 function(nanobind_extension_abi3 name)
   get_filename_component(ext "${NB_SUFFIX}" LAST_EXT)
-  set_target_properties(${name} PROPERTIES PREFIX "" SUFFIX ".abi3${ext}")
+  if(WIN32)
+    set_target_properties(${name} PROPERTIES PREFIX "" SUFFIX ".abi3${ext}")
+  else()
+    set_target_properties(${name} PROPERTIES PREFIX "" SUFFIX "${ext}")
```
Comment (wjakob) [inline cmake/nanobind-config.cmake:201]: I believe aDid you leave out a period here? (before `${ext}`)

**Comment by wjakob** [comment]
My proposed version: #222

File: cmake/nanobind-config.cmake [hunk 2]
```diff
@@ -195,7 +195,11 @@ endfunction()
 
 function(nanobind_extension_abi3 name)
   get_filename_component(ext "${NB_SUFFIX}" LAST_EXT)
-  set_target_properties(${name} PROPERTIES PREFIX "" SUFFIX ".abi3${ext}")
+  if(WIN32)
+    set_target_properties(${name} PROPERTIES PREFIX "" SUFFIX ".abi3${ext}")
```
Comment (henryiii) [inline cmake/nanobind-config.cmake:199]: I got this exactly backwards. 🤦

Comment (henryiii) [reply cmake/nanobind-config.cmake:199]: Going to go ahead and flip these so I can test scikit-build-core, will switch to #222 when it's ready.

**Pull request closed without merging**